<a href="https://colab.research.google.com/github/kazishuvo22/BigMart_Sales_Prediction/blob/master/BigMart_predict.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# **BigMart Sales prediction Dataset**

---

Find out the properties of a product, and store which impacts the sales of a product.

---




In [20]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

from google.colab import drive
import glob
%matplotlib inline


In [27]:
train_file="/content/drive/My Drive/cse445 dataset/Train.csv"
test_file="/content/drive/My Drive/cse445 dataset/Test.csv"
submission_file="/content/drive/My Drive/cse445 dataset/Submission.csv"

In [28]:
train_file, test_file, submission_file

('/content/drive/My Drive/cse445 dataset/Train.csv',
 '/content/drive/My Drive/cse445 dataset/Test.csv',
 '/content/drive/My Drive/cse445 dataset/Submission.csv')

In [30]:
train = pd.read_csv(train_file)
test = pd.read_csv(test_file)
submission = pd.read_csv(submission_file)

In [34]:
train

Item_Identifier  Item_Weight  ...        Outlet_Type  Item_Outlet_Sales
0              FDA15        9.300  ...  Supermarket Type1          3735.1380
1              DRC01        5.920  ...  Supermarket Type2           443.4228
2              FDN15       17.500  ...  Supermarket Type1          2097.2700
3              FDX07       19.200  ...      Grocery Store           732.3800
4              NCD19        8.930  ...  Supermarket Type1           994.7052
...              ...          ...  ...                ...                ...
8518           FDF22        6.865  ...  Supermarket Type1          2778.3834
8519           FDS36        8.380  ...  Supermarket Type1           549.2850
8520           NCJ29       10.600  ...  Supermarket Type1          1193.1136
8521           FDN46        7.210  ...  Supermarket Type2          1845.5976
8522           DRG01       14.800  ...  Supermarket Type1           765.6700

[8523 rows x 12 columns]

In [35]:
test

Item_Identifier  Item_Weight  ... Outlet_Location_Type        Outlet_Type
0              FDW58       20.750  ...               Tier 1  Supermarket Type1
1              FDW14        8.300  ...               Tier 2  Supermarket Type1
2              NCN55       14.600  ...               Tier 3      Grocery Store
3              FDQ58        7.315  ...               Tier 2  Supermarket Type1
4              FDY38          NaN  ...               Tier 3  Supermarket Type3
...              ...          ...  ...                  ...                ...
5676           FDB58       10.500  ...               Tier 1  Supermarket Type1
5677           FDD47        7.600  ...               Tier 3  Supermarket Type2
5678           NCO17       10.000  ...               Tier 2  Supermarket Type1
5679           FDJ26       15.300  ...               Tier 2  Supermarket Type1
5680           FDU37        9.500  ...               Tier 2  Supermarket Type1

[5681 rows x 11 columns]

In [36]:
submission

Unnamed: 0 Item_Identifier Outlet_Identifier  Item_Outlet_Sales
0              0           FDW58            OUT049        1636.244023
1              1           FDW14            OUT017        1409.355910
2              2           NCN55            OUT010         710.594286
3              3           FDQ58            OUT017        2355.184955
4              4           FDY38            OUT027        5857.916205
...          ...             ...               ...                ...
5676        5676           FDB58            OUT046        2125.984810
5677        5677           FDD47            OUT018        2602.671833
5678        5678           NCO17            OUT045        1832.451358
5679        5679           FDJ26            OUT017        3538.685188
5680        5680           FDU37            OUT045        1281.144462

[5681 rows x 4 columns]

## Combine test and train into one file

To explore data in any data science competetion, it is advisable to append test data to the train data. So, we will combine both train and test to carry out data visualization, feature engineering, one-hot encoding, and label encoding. Later we would split this combined data back to train and test datasets.








In [44]:
train['source']='train'
test['source']='test'
data = pd.concat([train, test],ignore_index=True)
print(train.shape, test.shape, data.shape)

(8523, 13) (5681, 12) (14204, 13)


train dataset has 8523 rows and 12 features and test has 5681 rows and 11 columns. train has 1 extra column which is the target variable. We will predict this target variable for the test dataset later.

In [45]:
data.dtypes

Item_Identifier               object
Item_Weight                  float64
Item_Fat_Content              object
Item_Visibility              float64
Item_Type                     object
Item_MRP                     float64
Outlet_Identifier             object
Outlet_Establishment_Year      int64
Outlet_Size                   object
Outlet_Location_Type          object
Outlet_Type                   object
Item_Outlet_Sales            float64
source                        object
dtype: object

In [46]:
data.head()

Item_Identifier  Item_Weight  ... Item_Outlet_Sales  source
0           FDA15         9.30  ...         3735.1380   train
1           DRC01         5.92  ...          443.4228   train
2           FDN15        17.50  ...         2097.2700   train
3           FDX07        19.20  ...          732.3800   train
4           NCD19         8.93  ...          994.7052   train

[5 rows x 13 columns]

##Numerical data summary:

In [47]:
data.describe()

Item_Weight  ...  Item_Outlet_Sales
count  11765.000000  ...        8523.000000
mean      12.792854  ...        2181.288914
std        4.652502  ...        1706.499616
min        4.555000  ...          33.290000
25%        8.710000  ...         834.247400
50%       12.600000  ...        1794.331000
75%       16.750000  ...        3101.296400
max       21.350000  ...       13086.964800

[8 rows x 5 columns]

## Data Cleaning
Missing data can have a severe impact on building predictive models because the missing values might be contain some vital information which could help in making better predictions. So, it becomes imperative to carry out missing data imputation.



### Check missing values:

In [59]:
data.apply(lambda x: sum(x.isnull()))

Item_Identifier                 0
Item_Weight                  2439
Item_Fat_Content                0
Item_Visibility                 0
Item_Type                       0
Item_MRP                        0
Outlet_Identifier               0
Outlet_Establishment_Year       0
Outlet_Size                  4016
Outlet_Location_Type            0
Outlet_Type                     0
Item_Outlet_Sales            5681
source                          0
dtype: int64


### Filling missing values


In [63]:
data['Item_Outlet_Sales'].value_counts()

2181.288914    5681
958.752000       17
1342.252800      16
1845.597600      15
703.084800       15
               ... 
4572.048600       1
2668.526400       1
7833.802800       1
8508.924000       1
158.460400        1
Name: Item_Outlet_Sales, Length: 3494, dtype: int64

In [64]:
data.Item_Outlet_Sales = data.Item_Outlet_Sales.fillna(data.Item_Outlet_Sales.mean())

In [68]:
data.Item_Outlet_Sales 

0        3735.138000
1         443.422800
2        2097.270000
3         732.380000
4         994.705200
            ...     
14199    2181.288914
14200    2181.288914
14201    2181.288914
14202    2181.288914
14203    2181.288914
Name: Item_Outlet_Sales, Length: 14204, dtype: float64

In [65]:
data['Item_Weight'].value_counts()

12.792854    2439
17.600000     135
12.150000     127
10.500000     123
13.650000     115
             ... 
8.380000        7
7.145000        7
9.035000        6
4.615000        6
7.850000        6
Name: Item_Weight, Length: 416, dtype: int64

In [69]:
data.Item_Weight = data.Item_Weight.fillna(data.Item_Weight.mean())
data.Item_Weight

0         9.30
1         5.92
2        17.50
3        19.20
4         8.93
         ...  
14199    10.50
14200     7.60
14201    10.00
14202    15.30
14203     9.50
Name: Item_Weight, Length: 14204, dtype: float64